Firestore/core/src/api/query_core.cc

```diff
@@ -163,7 +163,7 @@ std::unique_ptr<ListenerRegistration> Query::AddSnapshotListener(
       QuerySnapshot result(firestore_, query_, std::move(snapshot),
                            std::move(metadata));
 
-      user_listener_->OnEvent(result);
+      user_listener_->OnEvent(std::move(result));
```

Comment:
Is this actually related to the fix? Or is this just an optimization you discovered while implementing this fix?